### ISSUE TITLE
[Bug]: apps.plugin fails to collect stats

### ISSUE THREAD

**Issue by OliverNChalk** [netdata/netdata#14987]
### Bug description

I was told to open a report here after seeking further info on discord. I am getting these logs when doing `grep -i apps /var/log/netdata/error.log`:
```txt
2023-04-29 14:50:55: netdata INFO  : PD[apps] : thread created with task id 1348578
2023-04-29 14:50:55: netdata INFO  : PD[apps] : set name of thread 1348578 to PD[apps]
2023-04-29 14:50:55: netdata INFO  : PD[apps] : PLUGINSD: 'host:tokyo-aws-prod' connected to '/usr/libexec/netdata/plugins.d/apps.plugin' running on pid 1348604
2023-04-29 14:50:55: netdata ERROR : PD[apps] : PARSER: read failed: end of file
2023-04-29 14:50:55: netdata INFO  : PD[apps] : PLUGINSD: 'host:tokyo-aws-prod', '/usr/libexec/netdata/plugins.d/apps.plugin' (pid 1348604) disconnected after 0 successful data collections (ENDs).
2023-04-29 14:50:55: netdata ERROR : PD[apps] : child pid 1348604 exited with code 126.
2023-04-29 14:50:55: netdata ERROR : PD[apps] : PLUGINSD: 'host:tokyo-aws-prod', '/usr/libexec/netdata/plugins.d/apps.plugin' (pid 1348604) exited with error code 126 and haven't collected any data. Disabling it.
2023-04-29 14:50:55: netdata INFO  : PD[apps] : thread with task id 1348578 finished
```

Additionally, I am seeing similar errors for `ebpf`:
```
2023-04-29 14:50:55: netdata INFO  : PD[ebpf] : thread created with task id 1348569
2023-04-29 14:50:55: netdata INFO  : PD[ebpf] : set name of thread 1348569 to PD[ebpf]
2023-04-29 14:50:55: netdata INFO  : PD[ebpf] : PLUGINSD: 'host:tokyo-aws-prod' connected to '/usr/libexec/netdata/plugins.d/ebpf.plugin' running on pid 1348572
2023-04-29 14:50:55: netdata ERROR : PD[ebpf] : PARSER: read failed: end of file
2023-04-29 14:50:55: netdata INFO  : PD[ebpf] : PLUGINSD: 'host:tokyo-aws-prod', '/usr/libexec/netdata/plugins.d/ebpf.plugin' (pid 1348572) disconnected after 0 successful data collections (ENDs).
2023-04-29 14:50:55: netdata ERROR : PD[ebpf] : child pid 1348572 exited with code 3.
2023-04-29 14:50:55: netdata ERROR : PD[ebpf] : PLUGINSD: 'host:tokyo-aws-prod', '/usr/libexec/netdata/plugins.d/ebpf.plugin' (pid 1348572) exited with error code 3 and haven't collected any data. Disabling it.
2023-04-29 14:50:55: netdata INFO  : PD[ebpf] : thread with task id 1348569 finished
```

Checking systemd journal shows:
```txt
Apr 29 14:50:55 tokyo-aws-prod systemd[1]: Started Real time performance monitoring.
Apr 29 14:50:55 tokyo-aws-prod ebpf.plugin[1348572]: Does not have a configuration file inside `/etc/netdata/ebpf.d.conf. It will try to load stock file.
Apr 29 14:50:55 tokyo-aws-prod ebpf.plugin[1348572]: Loaded config file '/etc/netdata/apps_groups.conf'
Apr 29 14:50:55 tokyo-aws-prod ebpf.plugin[1348572]: ebpf.plugin should either run as root (now running with uid 115, euid 115) or have special capabilities..
```

So it seems I have insufficient priviliges for Netadata. The docs recommend running `ebpf` as root, however, I cannot seem to see a reference as to how this can be achieved? Should I be changing the systemd definition/using a drop-in to run the whole netadata app as root? If so, will it drop permissions once it's setup the requisite workers?

Sorry if this question has been answered in the docs but I'm a little lost here, the magic install while convenient is a little tricky to follow how the system is laid out post install.

### Expected behavior

Upon enabling apps.plugin, I was expecting to see apps metrics in the dashboard. There is no apps section and the error.log reports the plugin exiting on start.

### Steps to reproduce

1. Install via kickstart.sh on AWS Ubuntu 22.04 EC2
2. Enable apps.plugin
3. Errors/warnings around file reads & permissions
...

### Installation method

kickstart.sh

### System info

```shell
Linux tokyo-aws-prod 5.19.0-1022-aws #23~22.04.1-Ubuntu SMP Fri Mar 17 15:38:24 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
/etc/lsb-release:DISTRIB_ID=Ubuntu
/etc/lsb-release:DISTRIB_RELEASE=22.04
/etc/lsb-release:DISTRIB_CODENAME=jammy
/etc/lsb-release:DISTRIB_DESCRIPTION="Ubuntu 22.04.2 LTS"
/etc/os-release:PRETTY_NAME="Ubuntu 22.04.2 LTS"
/etc/os-release:NAME="Ubuntu"
/etc/os-release:VERSION_ID="22.04"
/etc/os-release:VERSION="22.04.2 LTS (Jammy Jellyfish)"
/etc/os-release:VERSION_CODENAME=jammy
/etc/os-release:ID=ubuntu
/etc/os-release:ID_LIKE=debian
/etc/os-release:UBUNTU_CODENAME=jammy
```

### Netdata build info

```shell
Version: netdata v1.38.0-432-nightly
Configure options:  '--build=x86_64-linux-gnu' '--includedir=${prefix}/include' '--mandir=${prefix}/share/man' '--infodir=${prefix}/share/info' '--disable-option-checking' '--disable-silent-rules' '--libdir=${prefix}/lib/x86_64-linux-gnu' '--libexecdir=${prefix}/lib/x86_64-linux-gnu' '--disable-maintainer-mode' '--prefix=/usr' '--sysconfdir=/etc' '--localstatedir=/var' '--libdir=/usr/lib' '--libexecdir=/usr/libexec' '--with-user=netdata' '--with-math' '--with-zlib' '--with-webdir=/var/lib/netdata/www' '--disable-dependency-tracking' 'build_alias=x86_64-linux-gnu' 'CFLAGS=-g -O2 -ffile-prefix-map=/usr/src/netdata=. -fstack-protector-strong -Wformat -Werror=format-security' 'LDFLAGS=-Wl,-Bsymbolic-functions -Wl,-z,relro' 'CPPFLAGS=-Wdate-time -D_FORTIFY_SOURCE=2' 'CXXFLAGS=-g -O2 -ffile-prefix-map=/usr/src/netdata=. -fstack-protector-strong -Wformat -Werror=format-security'
Install type: binpkg-deb
    Binary architecture: x86_64
    Packaging distro:
Features:
    dbengine:                   YES
    Native HTTPS:               YES
    Netdata Cloud:              YES
    ACLK:                       YES
    TLS Host Verification:      YES
    Machine Learning:           YES
    Stream Compression:         YES
Libraries:
    protobuf:                YES (system)
    jemalloc:                NO
    JSON-C:                  YES
    libcap:                  NO
    libcrypto:               YES
    libm:                    YES
    tcalloc:                 NO
    zlib:                    YES
Plugins:
    apps:                    YES
    cgroup Network Tracking: YES
    CUPS:                    YES
    EBPF:                    YES
    IPMI:                    YES
    NFACCT:                  YES
    perf:                    YES
    slabinfo:                YES
    Xen:                     NO
    Xen VBD Error Tracking:  NO
Exporters:
    AWS Kinesis:             NO
    GCP PubSub:              NO
    MongoDB:                 NO
    Prometheus Remote Write: YES
Debug/Developer Features:
    Trace Allocations:       NO
```

**Comment by ilyam8** [comment]
Hey, @OliverNChalk.

Can you show

```bash
ls -l /usr/libexec/netdata/plugins.d/apps.plugin
sudo getcap /usr/libexec/netdata/plugins.d/apps.plugin
```

And try to run the plugin in debug mode

```bash
# become netdata user
sudo su -s /bin/bash netdata

# run apps.plugin in debug mode
/usr/libexec/netdata/plugins.d/apps.plugin
```

**Comment by OliverNChalk** [comment]
```
ubuntu@tokyo-aws-prod:~$ ls -l /usr/libexec/netdata/plugins.d/apps.plugin
-rwxr-x--- 1 root root 848832 Apr 26 00:18 /usr/libexec/netdata/plugins.d/apps.plugin
ubuntu@tokyo-aws-prod:~$ sudo getcap /usr/libexec/netdata/plugins.d/apps.plugin
```
2nd command had no output.

```
ubuntu@tokyo-aws-prod:~$ sudo su -s /bin/bash netdata
netdata@tokyo-aws-prod:/home/ubuntu$ /usr/libexec/netdata/plugins.d/apps.plugin
bash: /usr/libexec/netdata/plugins.d/apps.plugin: Permission denied
```

On another server I performed a full uninstall then the following process resulted in a working build:
```
sudo /tmp/netdata-kickstart.sh --repositories-only
sudo apt install netdata
```

I think this issue could be related to running the full script & install root via sudo. In the latter process i simply use sudo to update the repositories then perform a more traditional install on Ubuntu.

**Comment by OliverNChalk** [comment]
Btw, happy for this issue to be closed if nothing obvious jumps out from my report.

**Comment by ilyam8** [comment]
@OliverNChalk I think the problem is the wrong group permission for apps.plugin. It is `root root` in your case, but should be `root netdata`.

---

I can't reproduce it. Used the following command:

> sudo ./kickstart.sh --disable-telemetry

And have correct permissions.


> I think this issue could be related to running the full script & install root via sudo. 

If you give the exact commands you used it would be helpful.

**Comment by OliverNChalk** [comment]
I forked from the ansible repo for setting up netdata, these are the tasks that run to set it up:

```yaml
---
- name: Download the installation script
  ansible.builtin.copy:
    src: netdata-kickstart.sh
    dest: /tmp/netdata-kickstart.sh
    mode: +x

- name: Install Netdata
  command: /tmp/netdata-kickstart.sh --dont-wait
  become: true

- name: Cleanup installation script
  file:
    path: /tmp/netdata-kickstart.sh
    state: absent
```

I have a local copy of the kickstart script checked in (file ending changed for github upload):

[netdata-kickstart.sh.txt](https://github.com/netdata/netdata/files/11374356/netdata-kickstart.sh.txt)

I ran this against an AWS intel x86 cpu VM running Ubuntu 22.04.

If you're unable to reproduce this I'm happy for the issue to be closed as it could have been an issue from an earlier install I ran that was not properly cleaned up. I usually file issues with a minimal repro cause I understand how tricky this can be. If I ever encounter this issue again ill try and isolate a minimal repro for you
